core/completion.py

```diff
@@ -575,11 +576,10 @@ def Matches(self, comp):
     executables = []  # type: List[str]
     for d in path_dirs:
       try:
-        st = posix.stat(d)
+        key = pyos.MakeDirCacheKey(d)
```

Comment:
I also wonder if Dict[Tuple2] will cause problems at runtime ... The issue is whether Tuple2 equality is implemented.

I think there was a bug about that in the past , but I guess we'll see

The problem is that we have some ad hoc C++ specializations, I think in `mycpp/comparators.h`

Yeah we might need `keys_equal()` for `Tuple2[str, int]` there?    I haven't looked at the code in awhile

Reply:
Oh good question. Wasn't sure what the state of affairs was here. I did do a manual check to be sure `Dict<Tuple<..>, ...>` compiled at least and things looked OK. I'll take a look at the comparators later today to be sure